src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -335,7 +335,7 @@ def _prepare_outputs_training_batch_end(
     ) -> Union[List[List[Dict[str, Any]]], List[Dict[str, Any]]]:
         """Processes the outputs from the batch loop into the format passed to the ``on_train_batch_end`` hook."""
         if not batch_output:
-            return []
+            return cast(List[Dict[str, Any]], [])
```

Comment:
This shouldn't be necessary, because an empty list still conforms with `List[Dict[str, Any]]`

Reply:
You mean updating `Union[List[List[Dict[str, Any]]], List[Dict[str, Any]]]` to `Union[List[List[Dict[str, Any]]], List[Dict[str, Any]], List]`? I think it is equivalent to `List[Any]` and it's too generic. :eyes:

Reply:
@carmocca what is your preference? I don't have a good answer what the best solution is.

Reply:
Since this is caused by a bug in mypy, I suggest ignoring the error (ref https://stackoverflow.com/a/61281305)

`type: ignore[return-value]`